### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by Geal**
some dependency updates require more invloved changes. Here LRU now requires a NonZeroUsize for the capacity argument. This changes the minimal value in the configuration file. Do we consider that this is a breaking change?

### CODE REVIEW

File: apollo-router/src/plugins/telemetry/apollo.rs [hunk 1]
```diff
@@ -136,8 +137,8 @@ const fn client_version_header_default() -> HeaderName {
     HeaderName::from_static(client_version_header_default_str())
 }
 
-pub(crate) const fn default_buffer_size() -> usize {
-    10000
+pub(crate) const fn default_buffer_size() -> NonZeroUsize {
+    unsafe { NonZeroUsize::new_unchecked(10000) }
```
Comment (garypen) [inline apollo-router/src/plugins/telemetry/apollo.rs:141]: I really dislike this api, but ...

**Review by garypen — COMMENTED** [review]
I think we could just make the configuration change from usize to NonZeroUsize and not document it. There are much more substantial changes taking place to config and we are calling them "compatible".

File: apollo-router/src/plugins/telemetry/apollo.rs (hunk 1 above)
Comment (Geal) [reply apollo-router/src/plugins/telemetry/apollo.rs:141]: there should be a panicking const constructor for `NonZeroUsize`, but there is not :(

**Comment by Geal** [comment]
my view here is that it does not really make sense to set the capacity to zero. I'm not sure there are cases where we would want to disable the cache entirely, except when debugging, and even then I have not felt the need for it

File: apollo-router/src/cache/mod.rs [hunk 2]
```diff
@@ -14,7 +15,7 @@ mod redis;
 pub(crate) mod storage;
 
 type WaitMap<K, V> = Arc<Mutex<HashMap<K, broadcast::Sender<V>>>>;
-pub(crate) const DEFAULT_CACHE_CAPACITY: usize = 512;
+pub(crate) const DEFAULT_CACHE_CAPACITY: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(512) };
```
[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/cache/mod.rs]: Nice one

**Review by bnjjj — APPROVED** [review]

**Pull request merged**